include/Core/UI/Component/AddonController.php

```diff
@@ -25,6 +25,55 @@ public function __construct( $file ) {
 	}
 
 	/**
+	 * Retrieves and outputs the Addon list HTML.
+	 *
+	 * return voids
+	 */
+	public function display_addons() {
+		// The products endpoint does not need a key or token to render published products.
+		// @todo: Change the API Url to get the actual addons.
+		$response = wp_remote_get( 'http://local.wordpress.dev/edd-api/products/' );
+
+		if ( is_wp_error( $response ) ) {
```

Comment:
- Seperate out the code that retrieves the addons and the code that displays it. You can use `render_addons` to display the addons and the change this method to `get_addons`, which will only retrieve the add-ons. Even better would be another function `render_addon` that takes a single add-on object and renders it.
- We need to seperate out the methods so that we can cache the response from the server. We don't want to hit our server on every page load. Cache the results for about 12 hours.
- Don't dump the error message to the user. Just have a generic message that says `We are not able to retrieve the add-on list now. Visit add-on page link to view the add-ons`